theatre/studio/src/utils/invariant.ts

```diff
@@ -0,0 +1,72 @@
+import {stringify} from './stringify'
```

Comment:
I do not understand this file. The implementations and types are terse and I don't understand how it is intended to be used.

Reply:
This is an invariant helper [see examples](https://javascript.hotexamples.com/examples/tiny-invariant/-/default/javascript-default-function-examples.html).

It is a really useful building block which provides its own Error which can clean up it's stack trace correctly for cases like in test files. All these helpers are really valuable carry-overs from another large codebase, but there's a lot to talk about with why these are each large QoL improvements over out-of-the box tools seen in jest/built-in functions.